src/mapgen_functions.h

```diff
@@ -46,6 +46,12 @@ struct mapgendata
   bool is_groundcover(const ter_id iid ) const;
 };
 
+/**
+ * Calculates the coordinates of a rotated point.
+ * Should match the `mapgen_*` rotation.
+ */
```

Comment:
> Though I'm not sure if it would be the best location for the description. After all this rotation is applied to all mapgen entries that have rotation.

Anywhere is good. Having it at one place allows you to point to it from wherever you use the same or similar logic.

--

It's now a public (in a header) function and can be used by *anyone* from *anywhere*, there is no indication that it's mapgen specific code. Therefor a comment regarding its behavior should be in order. Alternatively add `//mapgen specific code, don't use until you've read and understood its implementation` (it's not good, but it avoids implicit assumptions that will be wrong).

Btw. "Calculates the coordinates of a rotated point." for a function named `rotate_point` is kind of redundant. Yes, it "calculates" (nearly every piece of code does that) the "coordinates of a ... point" (the `point` parameter and the `point` return type hint that) and it applies some rotation (which is literally contained in the function name).